server/sonar-main/src/main/java/org/sonar/application/health/SearchNodeHealthProvider.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.application.health;
+
+import org.sonar.NetworkUtils;
+import org.sonar.api.utils.System2;
+import org.sonar.application.cluster.ClusterAppState;
+import org.sonar.cluster.health.NodeDetails;
+import org.sonar.cluster.health.NodeHealth;
+import org.sonar.cluster.health.NodeHealthProvider;
+import org.sonar.process.ProcessId;
+import org.sonar.process.Props;
+
+import static org.sonar.cluster.ClusterProperties.CLUSTER_NODE_NAME;
+import static org.sonar.cluster.ClusterProperties.CLUSTER_NODE_PORT;
+
+public class SearchNodeHealthProvider implements NodeHealthProvider {
+  private final ClusterAppState clusterAppState;
+  private final NodeDetails nodeDetails;
+
+  public SearchNodeHealthProvider(Props props, System2 system2, ClusterAppState clusterAppState, NetworkUtils networkUtils) {
+    this.clusterAppState = clusterAppState;
+    this.nodeDetails = NodeDetails.newNodeDetailsBuilder()
+      .setType(NodeDetails.Type.SEARCH)
+      .setName(props.nonNullValue(CLUSTER_NODE_NAME))
+      .setHost(networkUtils.getHostname())
```

Comment:
It should be the value of `sonar.cluster.host` as configured in sonar.properties. The localhost hostname does not make sense in response and may differ from configured IP.

Reply:
I discussed this with @christophelevis  and we agreed on this for the moment.
`sonar.cluster.node.host` is:
* optional
* can have multiple values

In these 2 cases, it does not meet the need of identifying the node => getHostName